src/L6/datasorter_test.py

```diff
@@ -45,9 +49,22 @@ def test_set_input_data_with_valid_paths(self):
         self.assertEqual(3, len(self.under_test.data))
         self.assertEqual(3, self.under_test.data[2])
 
+    def test_set_output_data_with_invalid_data(self):
+        with self.assertRaises(EmptyDataArrayException):
+            self.under_test.data = []
+            self.under_test.set_output_data(TEST_FILE_4)
+
+    def test_set_output_data_with_valid_data(self):
+        self.under_test.set_input_data(TEST_FILE_2)
+        data = self.under_test.data
+        self.assertTrue(self.under_test.set_output_data(TEST_FILE_4))
```

Comment:
More than checking that this is true, we need to verify that the file was created and that it contains all of the elements in `data`.

I suggest opening the files and parsing them using the `csv` module like you did in the reading method and checking that the list from the CSV is equal in size to `data` and that it contains all of the elements in `data`.